src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService service;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setPassword("123456");
+        user.setLogin("login1");
+        user.setAge(20);
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_OneUser_IsOk() {
+        service.register(user);
+        int expected = 1;
+        int actual = Storage.people.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_LargeDataInput_IsOk() {
+        user.setAge(99);
+        user.setPassword("1112284623846822");
+        user.setLogin("longExampleLogin_qwertyuiopasdfghjkl");
+        service.register(user);
+        int expected = 1;
+        int actual = Storage.people.size();
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_NullLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void register_NullPassword_NotOk() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void register_NullAge_NotOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void register_UnderMinValue_NotOk() {
+        user.setAge(6);
+        assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void register_UnderMinPasswordLength_NotOk() {
+        user.setPassword("1");
+        assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void register_UnderMinLoginLength_NotOk() {
+        user.setLogin("1");
+        assertThrows(RegistrationException.class, () -> service.register(user));
+    }
+
+    @Test
+    void register_sameLogin_NotOk() {
```

Comment:
```suggestion
    void register_SameLogin_NotOk() {
```
naming should be consistent